### PR TITLE
chore: upgrade deps to fix npm audit complaints

### PR DESCRIPTION
**Pull request by raymondfeng**
#### Related issues

- Recreated from https://github.com/strongloop/loopback-boot/pull/287.

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock=true
```
Comment (b-admike) [inline .npmrc]: Following from the other PR, can we remove package-lock file and this change now that we have the upgrades in `package.json`?

Comment (raymondfeng) [reply .npmrc]: I'm fine either way.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 sudo: false
 language: node_js
 node_js:
-  - "4"
```
Comment (raymondfeng) [inline .travis.yml]: Please note it won't work with Node 4.x any more. I find out `loopback@3.x` has `node >= 6`.

File: package.json [hunk 3]
```diff
@@ -20,29 +20,29 @@
...
   "license": "MIT",
   "dependencies": {
-    "async": "~0.9.0",
-    "commondir": "0.0.1",
-    "debug": "^2.0.0",
-    "lodash": "^4.17.5",
-    "semver": "^4.1.0",
-    "strong-globalize": "^2.6.2",
-    "toposort": "^0.2.10"
+    "async": "^2.6.2",
```
Comment (bajtos) [inline package.json]: IIRC, there are subtle but important changes in the way how individual callbacks are scheduled/executed between async versions 0.x and post-1.x.

Unless there is a know security vulnerability in `async@0.9.0`, we should not be upgrading this dependency.

Comment (raymondfeng) [reply package.json]: Reverted.

**Pull request merged**